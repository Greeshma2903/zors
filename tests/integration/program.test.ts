import { describe, expect, it } from 'vitest';
import { run } from '../helpers';

describe('Program()', () => {
  it('should print the program/command version', async () => {
    const cli = await run('', 'basic.ts', ['--v']);

    expect(cli.stdout).toMatchSnapshot();
  });

  it('should run the command action', async () => {
    const cli = await run('init', 'basic.ts', []);

    expect(cli.stdout).toMatchSnapshot();
  });

  it('should run with variadic args', async () => {
    const cli = await run('add', 'basic.ts', [
      'index.html',
      'app.js',
      'styles.css',
    ]);

    expect(cli.stdout).toMatchSnapshot();
  });

  it('should print help output', async () => {
    const cli = await run('', 'basic.ts', ['--h']);

    expect(cli.stdout).toMatchSnapshot();
  });
});


