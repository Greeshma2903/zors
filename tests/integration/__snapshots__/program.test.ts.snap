// Vitest Snapshot v1

exports[`Program() > should print help output 1`] = `""`;

exports[`Program() > should print the program/command version 1`] = `"GIT/v1.0.0"`;

exports[`Program() > should run the command action 1`] = `"Intialized an empty git repository"`;

exports[`Program() > should run with variadic args 1`] = `"Tracking files:  [ 'index.html', 'app.js', 'styles.css' ]"`;
